apps/src/redux.js

```diff
@@ -1,21 +1,124 @@
+/* eslint no-unused-vars: "error" */
 /** @file Code.org configured store-creation method.
  *  @see http://redux.js.org/docs/api/createStore.html */
 import Immutable from 'immutable';
 import experiments from "./util/experiments";
+import * as redux from 'redux';
+import reduxThunk from 'redux-thunk';
+import { connect } from 'react-redux';
 
-var redux = require('redux');
-var reduxThunk = require('redux-thunk').default;
 if (process.env.NODE_ENV !== "production") {
   var createLogger = require('redux-logger');
 }
 
+
+let reduxStore;
+let globalReducers = {};
+
+let __oldReduxStore;
+let __oldGlobalReducers;
+
+export function stubRedux() {
+  if (__oldReduxStore) {
+    throw new Error("Reduce store has already been stubbed. Did you forget to call restore?");
+  }
+  __oldReduxStore = reduxStore;
+  __oldGlobalReducers = globalReducers;
+  reduxStore = null;
+  globalReducers = {};
+}
+
+export function restoreRedux() {
+  reduxStore = __oldReduxStore;
+  globalReducers = __oldGlobalReducers;
+  __oldReduxStore = null;
+  __oldGlobalReducers = null;
+}
```

Comment:
that only works because `experiments` is the module under test. The problem is the following:

```jsx
// getSomething.js
import {getStore} from '@cdo/apps/redux';

export function getSomething() {
  return getStore().getState().some.state.object;
}
```

```jsx
// myAwesomeModule.js
import getSomething from './getSomething';

export function doSomethingAwesome() {
  console.log(getSomething());
}
```

```jsx
// myAwesomeModuleTest.js
import {doSomethingAwesome} from '@cdo/apps/myAwesomeModule';
let redux = require('@cdo/apps/redux');
describe('my awesome module', () => {
  beforeEach(() => {
    redux.getStore = /* some stub implementation that points to a different store instance */
  });
});
```

By the time we override `redux.getStore` inside our test suite, `getSomething.js` has already imported the original version of `getStore`. And furthermore, `getSomething.js` does not expose it's copy of `getStore` either so it can't be overridden later.

The only way to do this kind of module level stubbing/mocking properly is to hook into the module resolution/loading layer, the way we do with `@cdo/apps/locale`, but that is rather invasive. Some test frameworks ([jest](https://facebook.github.io/jest/)) do this automagically, but AFAIK, there isn't a good way to do this with karma/mocha/sinon.

I've used jest before, though it's been a while and it had a few "gotchas" that are a bit counterintuitive at first. We could if it's possible to use it piecemeal.